generation/tasks/RivetMerge.py

```diff
@@ -205,5 +210,96 @@ def run(self):
+        return reqs
+    
+    def run(self):
+
+        # ensure that the output directory exists
+        output = self.output()
+        try:
+            output.parent.touch()
+        except IOError:
+            logger.error("Output target doesn't exist!")
+
+        # actual payload:
+        print("=======================================================")
+        print("Starting merging of extension YODA files with base")
+        print("=======================================================")
+
+        # adjust rivet environment
+        dirname = os.path.expandvars("$ANALYSIS_PATH")
+        self.rivet_env["RIVET_ANALYSIS_PATH"] = ":".join((dirname, self.rivet_env.get("RIVET_ANALYSIS_PATH","")))
+
+        # identify and get the compiled Rivet analyses
+        logger.info("Shared object Rivet files: {}".format(self.input()["analyses"]["collection"].targets.values()))
+        local_so_files = []
+        for target in self.input()["analyses"]["collection"].targets.values():
+            with target.localize('r') as so_file:
+                local_path = os.path.join(dirname, str(os.path.basename(so_file.path)).split("_",1)[1])
+                so_file.copy_to_local(
+                    local_path
+                )
+                local_so_files.append(local_path)
+
+        # localize the separate YODA files on grid storage
+        inputfile_list = []
+        rivetMerge_dict = {k: v for k, v in self.input().items() if k.startswith("RivetMerge")}
+        for target in rivetMerge_dict.values():
+            if target.exists():
+                with target.localize('r') as _file:
+                    inputfile_list.append(_file.path)
+
+        # merge in chunks
+        chunk_size = self.chunk_size
+
+        final_input_files=inputfile_list
+        try:
+            if len(final_input_files) > 1:
+                while len(final_input_files)>chunk_size:
+                    final_input_files=self.mergeChunkwise(full_inputfile_list=final_input_files,chunk_size=chunk_size)
+                
+                _output_file = self.mergeSingleYodaChunk(inputfile_list=final_input_files)
+            else:
+                logger.info(f"No extensions met! Copying main campaign {self.campaign} RivetMerge output.")
+                _output_file = final_input_files[0]
+            _output_file = os.path.abspath(_output_file)
+
+            if os.path.exists(_output_file):
+                output.copy_from_local(_output_file)
+                os.remove(_output_file)
+                for _outfile in final_input_files:
+                    os.remove(_outfile)
+            else:
+                raise IOError("Could not find output file {}!".format(_output_file))
+        except Exception as e:
+            self.output().remove()
+            raise e
+        finally:
+            for so_file in local_so_files:
+                os.remove(so_file)
```

Comment:
Why are you removing the compiled analysis, if the merging of files is failing??

Reply:
Because they are stored on grid storage and get built for a certain build environment. This is however not reflected in the local file name on the executing machine. So whenever a local execution fails it is safer to remove everything, so a repeated execution in possibly a different environment does not fail because of incompatibility with a previously downloaded analysis file.